src/CFITSIO.jl

```diff
@@ -1036,6 +1081,8 @@ function fits_create_img(f::FITSFile, ::Type{T}, naxes::Vector{<:Integer}) where
         status,
     )
     fits_assert_ok(status[])
+    fits_get_img_size(f) == naxes || error("could not create the image correctly")
```

Comment:
This test takes around 80 ns with one allocation, I wonder if this assertion makes a significant difference?

Reply:
I didn't follow all the failures, and now I see none.  Care to elaborate what this comment is about?

Reply:
This assertion makes sure that the image created has the correct size. As such this shouldn't be necessary if all the `ccall`s are correct, but if the types passed to the `ccall` are not correct a corrupted image may be created. I had added this as a sanity check, but perhaps I should remove this now that things seem to work correctly.

Reply:
Incidentally this call to `fits_get_img_size` also appears to flush some internal buffer and make the image available for subsequent usage.

On master

```julia
julia> using CFITSIO

julia> dir = mktempdir()
"/tmp/jl_cw4TI7"

julia> filename = joinpath(dir, "temp.fits")
"/tmp/jl_cw4TI7/temp.fits"

julia> f = fits_clobber_file(filename)
FITSFile(Ptr{Nothing} @0x0000000002a38f00)

julia> a = ones(2,2);

julia> fits_create_img(f, eltype(a), [size(a)...]);

julia> fits_get_num_hdus(f)
0

julia> fits_get_img_size(f)
2-element Vector{Int64}:
 2
 2

julia> fits_get_num_hdus(f)
1
```

The image is not immediately written out after a call to `fits_create_img` (not sure what controls this). I have not seen this before, but evidently this may happen too. In such cases perhaps a call to `fits_get_img_size` here works as a buffer flush too, which is certainly desirable.

Reply:
I've reverted this for now to get back to the behavior on master.